service/src/main/java/datawave/microservice/querymetric/handler/ShardTableQueryMetricHandler.java

```diff
@@ -485,6 +488,31 @@ public List<T> getQueryMetrics(Query query) throws Exception {
         return queryMetrics;
     }
     
+    public QueryMetricsSubplanResponse subplan(String queryId) throws Exception {
+        QueryMetricsSubplanResponse response = new QueryMetricsSubplanResponse();
+        
+        Date end = new Date();
+        Date begin = DateUtils.setYears(end, 2000);
+        QueryImpl query = new QueryImpl();
+        query.setBeginDate(begin);
+        query.setEndDate(end);
+        query.setQueryLogicName(QUERY_METRICS_LOGIC_NAME);
+        // QueryMetricQueryLogic now enforces that you must be a QueryMetricsAdministrator to query metrics that do not belong to you
+        query.setQuery("QUERY_ID == '" + queryId + "'");
+        query.setQueryName(QUERY_METRICS_LOGIC_NAME);
+        query.setColumnVisibility(queryMetricHandlerProperties.getQueryVisibility());
+        query.setExpirationDate(DateUtils.addDays(new Date(), 1));
+        query.setPagesize(1000);
+        query.setId(UUID.randomUUID());
+        query.addParameter(QueryOptions.INCLUDE_GROUPING_CONTEXT, "true");
+        query.addParameter(QueryParameters.BLACKLISTED_FIELDS, "PAGE_METRICS");
```

Comment:
Consider avoiding duplicate code just to add BLACKLISTED_FIELDS.  
Maybe a pass additional Parameters to getQueryMetrics and keep this method small.

Also -- other ShardTableQueryMetric implementations use QUERY_ID == 'X' || OTHER_ID == 'X'.  That is why getQueryMetric(final String queryId) is abstracted out and relatively small.